pkg/hubble/container/ring_test.go

```diff
@@ -91,6 +93,79 @@ func BenchmarkTimeLibRingRead(b *testing.B) {
 	})
 }
 
+func BenchmarkReadBufferNoReaders(b *testing.B) {
+	benchmarkRingBuffer(b, benchmarkRingBufferOptions{})
+}
+
+func BenchmarkReadBufferOneReader(b *testing.B) {
+	benchmarkRingBuffer(b, benchmarkRingBufferOptions{
+		readers: 1,
+	})
+}
+
+func BenchmarkReadBufferEightReaders(b *testing.B) {
+	b.Skip() // FIXME
+	benchmarkRingBuffer(b, benchmarkRingBufferOptions{
+		readers: 8,
+	})
+}
+
+type benchmarkRingBufferOptions struct {
+	capacity capacity
+	readers  int
+}
+
+// benchmarkRingBuffer benchmarks RingBuffer's writes and reads with multiple
+// readers.
+func benchmarkRingBuffer(b *testing.B, options benchmarkRingBufferOptions) {
+	b.ReportAllocs()
+
+	if options.capacity == capacity(0) {
+		options.capacity = benchmarkCapacity
+	}
+
+	r := NewRing(options.capacity)
+
+	ctx, cancel := context.WithCancel(context.Background())
+
+	readerEventsReceived := make(chan int, options.readers)
+	for i := 0; i < options.readers; i++ {
+		rr := NewRingReader(r, 0)
+		go func() {
+			// Count the number of events received and report them over a
+			// channel.
+			eventsReceived := 0
+			for {
+				if rr.NextFollow(ctx) == nil {
+					readerEventsReceived <- eventsReceived
+					return
+				}
+				eventsReceived++
+			}
+		}()
+	}
+
+	event := &v1.Event{}
+
+	b.ResetTimer()
+
+	// Write the same event b.N times.
+	for i := 0; i < b.N; i++ {
```

Comment:
```suggestion
	for i := 0; i < b.N+1; i++ {
```
The last write is unfortunately never available for reading. Therefore, to read `1` event you need to write `2` (see also @gandro's comment and the issue referred in the comment).

Reply:
Thanks!